doc/tutorials/platformer/step_2.md

```diff
@@ -0,0 +1,164 @@
+- There should be pits that if Ember falls into, it is automatically game over.
+- There should be a main menu and a game-over screen that lets the player start over.
+
+Now that this is planned out, I know you are probably as excited as I am to begin and I just want to
+see Ember on the screen.  So let's do that first.
+
+
+## Loading Assets
+
+For Ember to be displayed, we will need to load the assets.  This can be done in `main.dart`, but by
+so doing, we will quickly clutter the file.  To keep our game organized, we should create files that
+have a single focus.  So let's create a file in the `lib` folder called `ember_quest.dart`.  In that
+file, we will add:
+
+```dart
+import 'package:flame/game.dart';
+
+class EmberQuestGame extends FlameGame {
+  EmberQuestGame();
+
+  @override
+  Future<void> onLoad() async {
+    await images.loadAll([
+      'block.png',
+      'ember.png',
+      'ground.png',
+      'heart_half.png',
+      'heart.png',
+      'star.png',
+      'water_enemy.png',
+    ]);
+
+  }
+}
+```
+
+As I mentioned in the [assets](step_1.md#assets) section, we are using multiple individual image
+files and for performance reasons, we should leverage Flames' built-in caching system which will
+only load the files once, but allow us to access them as many times as needed without an impact to
+the game.  `await images.loadAll()` takes a list of the file names that are found in `assets\images`
+and loads them to cache.
+
+
+## Scaffolding
+
+So now that we have our game file, let's prepare the `main.dart` file to receive our newly created
+`FlameGame`.  Change your entire `main.dart` file to the following:
+
+```dart
+import 'package:flame/game.dart';
+import 'package:flutter/material.dart';
+
+import 'ember_quest.dart';
+
+void main() {
+  runApp(
+    const GameWidget<EmberQuestGame>.controlled(
+      gameFactory: EmberQuestGame.new,
+    ),
+  );
+}
+```
+
+You can run this file and you should just have a blank screen now.  Let's get Ember loaded!
+
+
+## Ember Time
+
+Keeping your game files organized can always be a challenge.  I like to keep things logically
+organized by how they will be involved in my game.  So for Ember, let's create the following folder,
+`lib\actors` and in that folder, create `ember.dart`.  In that file, add the following code:
+
+```dart
+import 'package:flame/components.dart';
+
+import '../ember_quest.dart';
+
+class EmberPlayer extends SpriteAnimationComponent
+    with HasGameRef<EmberQuestGame> {
+  EmberPlayer({
+    required super.position,
+  }) : super(size: Vector2.all(64), anchor: Anchor.center);
+
+  @override
+  Future<void> onLoad() async {
+    animation = SpriteAnimation.fromFrameData(
+      gameRef.images.fromCache('ember.png'),
+      SpriteAnimationData.sequenced(
+        amount: 4,
+        textureSize: Vector2.all(16),
+        stepTime: 0.12,
+      ),
+    );
+  }
+}
+```
+
+This file uses the `HasGameRef` mixin which allows us to reach back to `ember_quest.dart` and
```

Comment:
`HasGameReference` is still in the experimental package, I think it's better to use the old mixin until we have replaced the old mixin completely.

Reply:
But the way to gradually replace anything is to start with documentation and tutorial: recommend people to use the new approach, and let the old one die out. This is a general remark, not necessarily about this particular use case. However, one of the main benefits of hosting tutorials on our own docs site (instead of publishing them, say, as blogs) is to be able to upgrade them as the API evolves.

Reply:
I can go either way, just let me know.